planex/cmd/createmockconfig.py

```diff
@@ -53,16 +93,20 @@ def load_mock_reference(fname):
     return config_opts
 
 
-def load_yum_repos(includes, excludes):
+def load_yum_repos(repo_config_list):
     """
     read in the yum repository configuration
     """
     yum_base = yum.YumBase()
-    enabled_set = {repo for pattern in includes
-                   for repo in yum_base.repos.findRepos(pattern)}
-    disabled_set = {repo for pattern in excludes
-                    for repo in yum_base.repos.findRepos(pattern)}
-    return list(enabled_set - disabled_set)
+    repo_set = {repo for repo in yum_base.repos.findRepos('*')
+                if repo.enabled}
+    for config, pattern in repo_config_list:
+        if config == "enable":
+            repo_set |= set(yum_base.repos.findRepos(pattern))
+        elif config == "disable":
+            repo_set -= set(yum_base.repos.findRepos(pattern))
+
+    return list(repo_set)
```

Comment:
This looks good.  Could you please write a couple of test cases to verify the behaviour?

Reply:
@euanh do you mean the unit test to cover those code changes?

Reply:
Yes.  It might be a bit tricky.   It should be possible to use mock (the mocking library, not the RPM builder!) or some other technique to override the yum_base object and return dummy repos.   Then run your method with different sets of options and verify that it works as you expect.

Reply:
That should work python mock can patch replace anything that you import.

Reply:
@euanh and @MarkSymsCtx 
Thanks.
I have updated the PR and create a test case for `createmockconfig.load_yum_repo`.
Please help to check.